server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -441,14 +443,24 @@ private static final class FailureCollector {
 
         List<FieldCapabilitiesFailure> build(Set<String> successfulIndices) {
             Map<Tuple<String, String>, FieldCapabilitiesFailure> indexFailures = new HashMap<>();
+            Set<String> exceptionsLogged = new HashSet<>();
             for (Map.Entry<String, Exception> failure : failuresByIndex.entrySet()) {
                 String index = failure.getKey();
                 Exception e = failure.getValue();
+                Throwable cause = ExceptionsHelper.unwrapCause(e);
+
+                // log 500 errors that are not Node/Shard unavailable type errors (and only once per Exception type)
+                if (ExceptionsHelper.status(e).getStatus() >= 500 && ExceptionsHelper.isNodeOrShardUnavailableTypeException(e) == false) {
+                    String loggingKey = e.getClass().getSimpleName() + (cause == null ? "" : cause.getClass().getSimpleName());
+                    if (exceptionsLogged.add(loggingKey)) {
+                        Supplier<String> messageSupplier = () -> String.format(Locale.ROOT, "Field caps Exception for index: [%s].", index);
+                        LOGGER.warn(messageSupplier, e);
```

Comment:
Curious to learn about advantages of calling logger this way. I'm usually just passing in the message directly, does the Supplier have advantages i.e. less resource usage etc...?

Reply:
I did that because I thought we have a precommit check that requires you to have a matching number of logging params (`{}`) as arguments and to get the stack track you don't want a `{}` param for it. But I just tried

```
                        LOGGER.warn("Field caps Exception for index: [{}]", index, e);
```

And the precommit command didn't complain, so I can change back to that, since it is less verbose.